vizro-core/tests/unit/vizro/models/_navigation/test_navigation.py

```diff
@@ -11,51 +12,59 @@
 
 @pytest.mark.usefixtures("dashboard_prebuild")
 class TestNavigationInstantiation:
-    """Tests navigation model instantiation."""
+    """Tests navigation model instantiation ."""
 
-    def test_navigation_default(self):
-        navigation = vm.Navigation(id="navigation")
-        assert navigation.id == "navigation"
-        assert navigation.pages == ["Page 1", "Page 2"]
+    @pytest.mark.parametrize("navigation", [None, vm.Navigation()])
+    def test_navigation_default(self, page1, page2, navigation):
+        # Navigation is optional inside Dashboard and navigation.pages will always be auto-populated if not provided
+        dashboard = vm.Dashboard(pages=[page1, page2], navigation=navigation)
+        assert hasattr(dashboard.navigation, "id")
+        assert dashboard.navigation.pages == ["Page 1", "Page 2"]
 
-    def test_navigation_pages_as_list(self, pages_as_list):
+    def test_navigation_valid_pages_as_list(self, pages_as_list):
         navigation = vm.Navigation(pages=pages_as_list, id="navigation")
         assert navigation.id == "navigation"
         assert navigation.pages == pages_as_list
 
-    def test_navigation_pages_as_dict(self, pages_as_dict):
+    def test_navigation_valid_pages_as_dict(self, pages_as_dict):
         navigation = vm.Navigation(pages=pages_as_dict, id="navigation")
         assert navigation.id == "navigation"
         assert navigation.pages == pages_as_dict
 
-    def test_navigation_not_all_pages_included(self):
-        with pytest.warns(UserWarning):
-            vm.Navigation(pages=["Page 1"])
+    def test_navigation_valid_pages_not_all_included(self):
```

Comment:
Similar, what does not all included mean here?

Reply:
Duplicate, will answer above